library/src/main/java/com/xwray/groupie/DiffTask.java

```diff
@@ -13,20 +13,25 @@
  * in a background thread. This task will call {@link AsyncDiffUtil.Callback#onDispatchResult(Collection)}
  * passing the new list just before dispatching the diff result to the provided
  * {@link DiffUtil.Callback} so that the new list.
- * <p>This task is executed via {@link AsyncDiffUtil#calculateDiff(Collection, DiffUtil.Callback)}.
+ * <p>This task is executed via {@link AsyncDiffUtil#calculateDiff(Collection, DiffUtil.Callback, OnAsyncUpdateListener)}.
  */
 class DiffTask extends AsyncTask<Void, Void, DiffUtil.DiffResult> {
     private final DiffUtil.Callback diffCallback;
     private final WeakReference<AsyncDiffUtil> asyncListDiffer;
     private final int runGeneration;
+    @Nullable private WeakReference<OnAsyncUpdateListener> onAsyncUpdateListener;
```

Comment:
AsyncTasks scare me and make me paranoid so I decided to wrap this in a `WeakReference` just to be safe. If anyone knows that it's safe not to use one, let me know

Reply:
You did right. It should be wrapped with WeakReference. The listener can potentially be an instance of/hold reference to Activity/Fragament. Since AsyncTasks don't support cancellation, WeakReference is the right tool to handle that.